File: R/vit-class-token.R
prepend_class_token <- function(patches, embed_dim) {
  batch_size <- dim(patches)[1]
  cls_token <- array(rnorm(embed_dim, sd = 0.02), dim = c(1, 1, embed_dim))
  cls_token_batch <- array(rep(cls_token, batch_size), dim = c(batch_size, 1, embed_dim))
  array(c(cls_token_batch, patches), dim = c(batch_size, dim(patches)[2] + 1, embed_dim))
}

File: Julia/vit-class-token.jl
function prepend_class_token(patches, embed_dim::Int)
    batch_size = size(patches, 1)
    cls_token = randn(1, 1, embed_dim) .* 0.02
    cls_token_batch = repeat(cls_token, batch_size, 1, 1)
    cat(cls_token_batch, patches; dims=2)
end
